apps/sms/test/marionette/mocks/mock_navigator_moz_mobile_message.js

```diff
@@ -43,12 +103,16 @@ Services.obs.addObserver(function(document) {
         storagePromise = appWindow.TestStorages.getStorage('messagesDB');
       }
 
-      storagePromise = storagePromise.then(function(storage) {
-        return storage || {
-          threads: new Map(),
-          recipientToThreadId: new Map(),
-          uniqueMessageIdCounter: 0
-        };
+      storagePromise = storagePromise.then((storage) => {
+        if (!storage) {
+          return {
+            threads: new Map(),
+            recipientToThreadId: new Map(),
+            uniqueMessageIdCounter: 0
+          };
+        }
+
+        return rehydrateAttachments(storage).then(() => storage);
       });
```

Comment:
also, maybe getStorage could reject if it finds no storage, in that case we could do:

```javascript
storagePromise = storagePromise.catch(() => {
   return {
       threads: new Map(),
       recipientToThreadId: new Map(),
       uniqueMessageIdCounter: 0
   };
}).then(rehydrateAttachments).then(() => storage);
```